src/Data/Nat/Binary/Minus.agda

```diff
@@ -0,0 +1,378 @@
+  (x' ∸ y') + y'                 ≡⟨ cong ((x' ∸ y') +_) (1+[2_]-suc-double y) ⟩
+  double x∸y + (suc (double y))  ≡⟨ x+suc[y]≡suc[x]+y (double x∸y) (double y) ⟩
+  suc (double x∸y) + double y    ≡⟨ suc-+ (double x∸y) (double y) ⟩
+  suc (double x∸y + double y)    ≡⟨ cong suc $ sym $ double-distrib-+ x∸y y ⟩
+  suc (double (x∸y + y))         ≡⟨ sym (1+[2_]-suc-double _) ⟩
+  1+[2 (x∸y + y) ]               ≡⟨ cong 1+[2_] $ sym $ x≥y⇒x≡[x∸y]+y (inj₁ y<x) ⟩
+  1+[2 x ]                       ∎
+  where open ≤-Reasoning;  x∸y = x ∸ y
+
+x≥y⇒x≡[x∸y]+y {x'@(1+[2 x ])} {1+[2 .x ]} (inj₂ refl) =  sym $ begin-equality
+  (x' ∸ x') + x'  ≡⟨ cong (_+ x') (x∸x≡0 x') ⟩
+  0ᵇ + x'         ≡⟨ sym (+-identityˡ x') ⟩
+  x'              ∎
+  where open ≤-Reasoning
+
+[x+y]∸y≡x :  ∀ x y → (x + y) ∸ y ≡ x
+[x+y]∸y≡x x y =  sym x≡d
+  where
+  d       = (x + y) ∸ y;           y≤x+y = x≤y+x y x
+  x+y≡d+y = x≥y⇒x≡[x∸y]+y y≤x+y;   x≡d   = +-cancelʳ-≡ x d x+y≡d+y
+
+[x+y]∸x≡y :  ∀ x y → (x + y) ∸ x ≡ y
+[x+y]∸x≡y x y =  trans (cong (_∸ x) (+-comm x y)) ([x+y]∸y≡x y x)
+
+∸-identityʳ :  RightIdentity 0ᵇ _∸_
+∸-identityʳ x =  trans (cong (_∸ 0ᵇ) $ sym $ +-identityʳ x) ([x+y]∸y≡x x 0ᵇ)
+
+x+[y∸x]≡y :  ∀ {x y} → x ≤ y → x + (y ∸ x) ≡ y
+x+[y∸x]≡y {x} {y} x≤y =  begin-equality
+  x + (y ∸ x)   ≡⟨ +-comm x _ ⟩
+  (y ∸ x) + x   ≡⟨ sym (x≥y⇒x≡[x∸y]+y x≤y) ⟩
+  y             ∎
+  where open ≤-Reasoning
+
+[x∸y]+y≡x :  ∀ {x y} → y ≤ x → (x ∸ y) + y ≡ x
+[x∸y]+y≡x {x} {y} y≤x =  trans (+-comm (x ∸ y) y) (x+[y∸x]≡y y≤x)
+
+toℕ-homo-∸ :  ∀ x y → toℕ (x ∸ y) ≡ (toℕ x) ℕ.∸ (toℕ y)
+toℕ-homo-∸ x y  with <-cmp x y
+... | tri< x<y _ _ =  begin
+  toℕ (x ∸ y)   ≡⟨ cong toℕ (x≤y⇒x∸y≡0 x≤y) ⟩
+  toℕ 0ᵇ        ≡⟨ refl ⟩
+  0             ≡⟨ sym (ℕₚ.m≤n⇒m∸n≡0 m≤n) ⟩
+  m ℕ.∸ n       ∎
+  where
+  open ≡-Reasoning
+  m = toℕ x;  n = toℕ y;  x≤y = <⇒≤ x<y;  m≤n = toℕ-mono-≤ x≤y
+
+... | tri≈ _ x≡y _ =  begin
+  toℕ (x ∸ y)  ≡⟨ cong toℕ (x≤y⇒x∸y≡0 {x} {y} x≤y) ⟩
+  toℕ 0ᵇ       ≡⟨ refl ⟩
+  0            ≡⟨ sym (ℕₚ.m≤n⇒m∸n≡0 m≤n) ⟩
+  m ℕ.∸ n      ∎
+  where
+  open ≡-Reasoning
+  m = toℕ x;   n = toℕ y;   x≤y = ≤-reflexive x≡y;   m≤n = toℕ-mono-≤ x≤y
+
+... | tri> _ _ y<x =  begin
+  toℕ (x ∸ y)                    ≡⟨ sym (ℕₚ.m+n∸n≡m (toℕ (x ∸ y)) n) ⟩
+  ((toℕ (x ∸ y)) ℕ.+ n) ℕ.∸ n    ≡⟨ cong (ℕ._∸ n) eq ⟩
+  ((m ℕ.∸ n) ℕ.+ n) ℕ.∸ n        ≡⟨ ℕₚ.m+n∸n≡m (m ℕ.∸ n) n ⟩
+  m ℕ.∸ n                        ∎
+  where
+  open ≡-Reasoning
+  m = toℕ x;   n = toℕ y;   y≤x = <⇒≤ y<x;   n≤m = toℕ-mono-≤ y≤x
+  eq = begin
+    toℕ (x ∸ y) ℕ.+ n    ≡⟨ sym (toℕ-homo-+ (x ∸ y) y) ⟩
+    toℕ ((x ∸ y) + y)    ≡⟨ cong toℕ (+-comm (x ∸ y) y) ⟩
+    toℕ (y + (x ∸ y))    ≡⟨ cong toℕ (x+[y∸x]≡y y≤x) ⟩
+    m                    ≡⟨ sym (ℕₚ.m+[n∸m]≡n n≤m) ⟩
+    n ℕ.+ (m ℕ.∸ n)      ≡⟨ ℕₚ.+-comm n (m ℕ.∸ n) ⟩
+    (m ℕ.∸ n) ℕ.+ n      ∎
+
+
+fromℕ-homo-∸ :  ∀ m n → fromℕ (m ℕ.∸ n) ≡ (fromℕ m) ∸ (fromℕ n)
+fromℕ-homo-∸ m n =  begin-equality
```

Comment:
Given that we have `fromℕ-toℕ`, it feels like we ought to be able to prove a more general
lemma. i.e.

```agda
  (f : ℕ → ℕ) (g : ℕᵇ → ℕᵇ) →
  (∀ m n → toℕ (g m n) ≡ f (toℕ m) (toℕ n)) →
  (∀ m n → fromℕ (f m n) ≡ g (fromℕ m) (fromℕ n))
```

and the converse.